rfcs/000-resource-notices.md

```diff
@@ -0,0 +1,88 @@
+
+## Guide-level Explanation (aka Product Details)
+
+An Amundser final user can see a notice in the table/dashboard/user detail page whenever his system admin adds one related to it.
+
+A notice is a small box with an icon and a message that can contain HTML markup (like links and bolded text).
+
+These will come in three flavors:
+
+- Informational: Marked with a blue "i" icon on the right side
+- Warning: Marked with an orange exclamation mark icon on the right side
+- Alert: Marked with a red exclamation mark icon on the right side
+
+For example, if company X wants to deprecate the use of one table, they can opt to add a new notice in their configuration file:
+
+```
+    ... Table Configuration
+    notices: {
+        "<SCHEMANAME>.<TABLENAME>": {
+          severity: NoticeSeverity.ALERT,
+          message: `<span>This table is deprecated, please use <a href="<LINKTONEWTABLEDETAILPAGE>">this new table</a> instead.</span>`,
+        },
+    },
+```
+
+This will show a notice with a red exclamation icon whenever a final user visits the Table Detail page of this table.
+
+The ultimate goal of this feature is to allow Amundsen administrators to point their users to more trusted/higher quality data without removing the old references.
+
+## Reference-level Explanation (aka Technical Details)
+
+The design we propose is based in the following:
+
+- Using our current frontend configuration files (config-utils.ts, config-default.ts, config-custom.ts and config-types.ts)
+- Allowing a three level "severity" schema for the alerts (info, warning, and alert)
+- Allowing its use in "tables" and "dashboards" (we could add it to people as well, but we'll need designs for it)
+
+What we are not supporting for now:
+
+- Multiple notifications per resource
+- Loading of external files (like a JSON)
+- Application of a single notice to multiple resources
+
+In the implementation, we'll simply check the notices object whenever we visit the resource detail page, showing any notification if present.
+
+It is not a breaking change, there won't be any costs for older versions to support this.
+
+## Drawbacks
+
+- This could not be scalable, as maybe users want to have too many notifications. It doesn't mean this work won't be a good start, but it could be not enough. TBD
+- With this approach, in order to add/update/remove a notification, admins would need to make a new deploy of the application. A better approach would be if we use a UI, so a proper feature flag system would be much better for this.
+- Notices are in a 1:1 basis. If admins wanted to deprecate a whole schema or dashboard product, they would need to create a notice for each table/dashboard.
+
+## Alternatives
```

Comment:
You already mentioned this in future possibilities, but I think the primary alternative to consider is sourcing this stuff from the graph itself. I think that would be preferable to feature flagging systems, since it's integrated into the graph and not tacked-on (unless i'm missing exactly how the feature flag system would be used here)

agreed to leave it out-of-scope for now

Reply:
You are probably right!

I guess the benefit of the feature flags would be to allow anybody to create/update/remove notifications.

Reply:
+1